terraform ssm association issue

_This issue was originally opened by @fdavier as hashicorp/terraform#16988. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
v0.11.1

### Terraform Configuration Files
**I added "aws_ssm_document" and "aws_ssm_association" after existing instance ressource:**
**My hope is to be able to launch commands to update/install additional features some linux instances after their creation**

```
resource "aws_instance" "gw0" {
  ami = "ami-d33ecdab"
  key_name = "infav10-devHub"
  instance_type = "m4.large"
  subnet_id = "${aws_subnet.public0.id}"
  associate_public_ip_address = true
  vpc_security_group_ids = ["${aws_security_group.gtw_security_group.id}"]
  private_ip = "10.0.2.11"

  root_block_device {
   volume_type = "gp2"
  }
  
  tags {
    Name = "${var.prefix_tag}-Gateway-0"
  }
      iam_instance_profile = "${aws_iam_instance_profile.gw-instance-profile.name}"
}
resource "aws_eip" "gw0" {
  vpc = true
  instance = "${aws_instance.gw0.id}"
  associate_with_private_ip = "10.0.2.11"
}

resource "aws_ssm_document" "gw_document" {
  name          = "gw_document"
  document_type = "Command"

  content = <<DOC
 {
   "schemaVersion":"2.0",
   "description":"Sample document",
   "mainSteps":[
      {
         "action":"aws:runPowerShellScript",
         "name":"runPowerShellScript",
         "inputs":{
            "runCommand":[
               "hostname"
            ]
         }
      }
   ]
}
DOC
}

resource "aws_ssm_association" "gw0" {
  name        = "gw_document"
  instance_id = "${aws_instance.gw0.id}"
}
```

### Debug Output

```
2017-12-26T09:43:38.674+0100 [DEBUG] plugin.terraform.exe: local-exec-provisione
r (internal) 2017/12/26 09:43:38 [DEBUG] plugin: waiting for all plugin processe
s to complete...
Error: Error applying plan:

1 error(s) occurred:

* aws_ssm_association.gw0: 1 error(s) occurred:

* aws_ssm_associati2017-12-26T09:43:38.701+0100 [DEBUG] plugin: plugin process e
xited: path=C:\tools\terraform\infa-hub\.terraform\plugins\windows_amd64\terrafo
rm-provider-aws_v0.1.4_x4.exe
on.gw0: [ERROR] Error c2017/12/26 09:43:38 [ERR] yamux: Failed to read stream da
ta: read tcp 127.0.0.1:54622->127.0.0.1:10000: wsarecv: Une connexion existante
a dû être fermée par l'hôte distant.
reat2017/12/26 09:43:38 [ERR] yamux: Failed to write header: write tcp 127.0.0.1
:54622->127.0.0.1:10000: use of closed network connection
2017-12-26T09:43:38.704+0100 [WARN ] plugin: error closing client during Kill: e
rr="unexpected EOF"
ing2017-12-26T09:43:38.705+0100 [DEBUG] plugin: plugin process exited: path=C:\t
ools\terraform\terraform.exe
 SSM association: InvalidDocument: Document schema version, 2.0, is not supporte
d by association that is created with instance id
        status code: 400, request id: ddef467a-ea18-11e7-988e-75f688c18213
```

### Expected Behavior
expect that Association is done and 'hostname' command executed on existing ec2 linux instance

### Actual Behavior
Don't execute, see crash output, ssm association don't work

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform plan`
3. `terraform apply`


### Important Factoids

### References

--- a/website/docs/r/ssm_association.html.markdown
+++ b/website/docs/r/ssm_association.html.markdown
@@ -60,7 +60,10 @@ DOC
 
 resource "aws_ssm_association" "foo" {
   name        = "test_document_association-%s"
-  instance_id = "${aws_instance.foo.id}"
+  targets {
+    key    = "InstanceIds"
+    values = "${aws_instance.foo.id}"
+  }
 }
 ```
 
@@ -71,7 +74,7 @@ The following arguments are supported:
 * `name` - (Required) The name of the SSM document to apply.
 * `association_name` - (Optional) The descriptive name for the association.
 * `document_version` - (Optional) The document version you want to associate with the target(s). Can be a specific version or the default version.
-* `instance_id` - (Optional) The instance ID to apply an SSM document to.
+* `instance_id` - (Optional) The instance ID to apply an SSM document to. Use `targets` with key `InstanceIds` for document schema versions 2.0 and above.
 * `output_location` - (Optional) An output location block. Output Location is documented below.
 * `parameters` - (Optional) A block of arbitrary string parameters to pass to the SSM document.
 * `schedule_expression` - (Optional) A cron expression when the association will be applied to the target(s).

--- a/website/docs/r/ssm_association.html.markdown
+++ b/website/docs/r/ssm_association.html.markdown
@@ -13,56 +13,12 @@ Associates an SSM Document to an instance or EC2 tag.
 ## Example Usage
 
 ```hcl
-resource "aws_security_group" "tf_test_foo" {
-  name        = "tf_test_foo"
-  description = "foo"
+resource "aws_ssm_association" "example" {
+  name = "${aws_ssm_document.example.name}"
 
-  ingress {
-    protocol    = "icmp"
-    from_port   = -1
-    to_port     = -1
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-}
-
-resource "aws_instance" "foo" {
-  # eu-west-1
-  ami               = "ami-f77ac884"
-  availability_zone = "eu-west-1a"
-  instance_type     = "t2.small"
-  security_groups   = ["${aws_security_group.tf_test_foo.name}"]
-}
-
-resource "aws_ssm_document" "foo_document" {
-  name          = "test_document_association-%s"
-  document_type = "Command"
-
-  content = <<DOC
-  {
-    "schemaVersion": "1.2",
-    "description": "Check ip configuration of a Linux instance.",
-    "parameters": {
-
-    },
-    "runtimeConfig": {
-      "aws:runShellScript": {
-        "properties": [
-          {
-            "id": "0.aws:runShellScript",
-            "runCommand": ["ifconfig"]
-          }
-        ]
-      }
-    }
-  }
-DOC
-}
-
-resource "aws_ssm_association" "foo" {
-  name        = "test_document_association-%s"
   targets {
     key    = "InstanceIds"
-    values = "${aws_instance.foo.id}"
+    values = "${aws_instance.example.id}"
   }
 }
 ```

